source/common/http/conn_manager_impl.h

```diff
@@ -691,6 +695,16 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
   const Server::OverloadActionState& overload_stop_accepting_requests_ref_;
   const Server::OverloadActionState& overload_disable_keepalive_ref_;
   TimeSource& time_source_;
+  using PerConnectionObjectMap =
+      absl::flat_hash_map<const std::string, PerConnectionObjectSharedPtr>;
+  // Function to get per_connection_object_map_
+  std::unique_ptr<PerConnectionObjectMap> per_connection_object_map_;
+  PerConnectionObjectMap* getPerConnectionObjectMap() {
```

Comment:
nit: return `PerConnectionObjectMap&`

Reply:
I will fix this

Reply:
I have fixed this. While I was at it; I tried to understand the motivation. I have scoured the sources online and find recommendations for and against such an use.

If my questions don't bother you much :) ... can you please help me understand why returning by reference here is better than returning a pointer?

Reply:
Because it can't be null.

Reply:
To prevent future changes to `getPerConnectionObjectMap` in returning nullptr; is what I understand. Thanks!